torch_xla/csrc/xla_lower_util.cpp

```diff
@@ -1228,4 +1228,11 @@ xla::XlaOp BuildCustomSharding(const xla::XlaOp& input) {
                          {input}, ShapeHelper::ShapeOfXlaOp(input));
 }
 
+xla::XlaOp BuildCustomMarkSharding(const torch::lazy::BackendDevice& device,
+                                   const xla::XlaOp& input,
+                                   const xla::XlaOp sharding) {
+  return xla::CustomCall(input.builder(), /*call_target_name=*/"MarkSharding",
```

Comment:
Could we re-use the existing `CustomSharding` op? `mark_sharding` can be translated into the `CustomSharding` op for an IR, which is the activation node. Unless it's the Dynamo side change to register the custom ops, we can reuse the existing `extern const OpKindWrapper xla_custom_sharding`, no?

Reply:
The original intention here was to have a separate custom op that includes all the logic for `mark_sharding` so Dynamo can properly capture `mark_sharding` as an op. All the logic currently at `xla_mark_sharding` at `init_python_bindings.cpp` would be moved to the lowering logic of this new custom op, so Dynamo can trace and recognize it properly.